cmd/genericclaimparams.go

```diff
@@ -129,9 +129,24 @@ func (sp *GenericClaimsParams) Run(ctx ActionCtx, claim jwt.Claims, r *store.Rep
 		}
 	}
 
-	cd.Tags.Add(sp.tags...)
-	cd.Tags.Remove(sp.rmTags...)
-	sort.Strings(cd.Tags)
+	var tags *jwt.TagList
+
+	switch claim.ClaimType() {
+	case jwt.OperatorClaim:
+		tags = &claim.(*jwt.OperatorClaims).Tags
+	case jwt.ActivationClaim:
+		tags = &claim.(*jwt.ActivationClaims).Tags
+	case jwt.AccountClaim:
+		tags = &claim.(*jwt.AccountClaims).Tags
+	case jwt.UserClaim:
+		tags = &claim.(*jwt.UserClaims).Tags
+	default:
+		panic("unhandled claim type")
```

Comment:
This is only when a programming error when someone using the control yet it is not one of the above... Not sure we should have panic but also think that it will be fine.